components/decorateMapComponent.js

```diff
@@ -24,41 +27,53 @@ export const contextTypes = {
   provider: PropTypes.string,
 };
 
-export const createNotSupportedComponent = message => () => console.error(message) || null;
+export const createNotSupportedComponent = message => () => {
+  console.error(message);
+  return null;
+};
 
-export const googleMapIsInstalled = !!NativeModules.UIManager[getAirMapName('google')];
+export const googleMapIsInstalled = !!NativeModules.UIManager[getAirMapName(PROVIDER_GOOGLE)];
 
 export default function decorateMapComponent(Component, { componentType, providers }) {
-  const components = {
-    default: requireNativeComponent(getAirComponentName(null, componentType), Component),
-  };
+  let components = {};
+
+  const getDefaultComponent = () => requireNativeComponent(getAirComponentName(null, componentType), Component);
+
+  Component.contextTypes = contextTypes;
+  Component.prototype.getAirComponent = function() {
+    const provider = this.context.provider || PROVIDER_DEFAULT;
```

Comment:
the problem with this is it will cause us to call `getDefaultComponent` 2x instead of 1x